src/warnet/warnet.py

```diff
@@ -220,3 +220,10 @@ def export(self, subdir):
         config_path = os.path.join(subdir, "sim.json")
         with open(config_path, "a") as f:
             json.dump(config, f)
+
+    def populate_ip_addresses(self):
+        while not all(tank.ipv4 is not None for tank in self.tanks):
```

Comment:
in "backend: remove ipv4 from compose backend" (https://github.com/bitcoin-dev-project/warnet/commit/50107b7f1dd88a1f0d393ed0e64b2ef8b19fddc7):

oof.. this is gonna be fun with 500 nodes. but not sure if there's a cleaner way

Reply:
I think I can make the loop cleaner.

Also, we should be able to fetch an entire `inspect network` (or equivalent), and parse that in a oner

Reply:
dunno about docker, but i think the k8s api gives us some nice endpoints that should give everything we need in one response

Reply:
On the compose side I at least cache the result of the first lookup, which seems to work.

https://github.com/bitcoin-dev-project/warnet/pull/254/files#diff-adeb9c7bdada05ccdf2c802012d7c32c0737f526b7ed1d34443e7678cb81236aR499-R516